File: python/webhooks/ping_webhook.py
import requests
from requests.exceptions import HTTPError

api_url = "https://api.theauthapi.com/webhooks"
access_key = "REPLACE_WITH_YOUR_ACCESS_KEY"

headers = {
    "x-api-key": access_key,
}

def ping_webhook(ping_webhook_data):
    try:
        response = requests.post(f'{api_url}/ping', json=ping_webhook_data, headers=headers)
        response.raise_for_status()
        return response.json()
    except HTTPError as error: 
        # handle error
        print(error)


ping_webhook_data = {
    'url': 'https://webhook.site/9f2b7979-f9ce-4ecc-b8a2-299c0b08d114',
    'httpMethod': 'POST',
    'customHeaders': {
        # custom headers added to the request sent to the webhook
    },
    'body': {
        'name': 'My first api key test',
        'key': 'random_key_12412569120',
    }
}
ping_webhook_response = ping_webhook(ping_webhook_data)
print(ping_webhook_response)

File: js/axios/webhooks/pingWebhook.js
import axios from "axios";
import "dotenv/config";

const accessKey = process.env.ACCESS_TOKEN;
const apiUrl = process.env.production
  ? "https://api.theauthapi.com"
  : process.env.TESTING_URL;

async function pingWebhook(webhookData) {
  try {
    const { data } = await axios
      .post(apiUrl + "/webhooks/ping", webhookData, {
        headers: {
          ContentType: "application/json",
          "x-api-key": accessKey,
        },
      });
    return data;
  } catch (error) {
    if (error.response) {
      console.log(error.response.data);
    } else {
      // handle other errors
    }
  }
}

const webhookData = {
  url: "https://webhook.site/a61857c1-ead8-4c2a-86a3-ebc910db2271",
  httpMethod: "POST", // http method of the request sent to the webhook
  customHeaders: {
    // custom headers added to the request sent to the webhook
    a_random_number: "WEBHOOK-UID-" + Math.random(),
  },
  body: { a_random_number: "WEBHOOK-UID-" + Math.random() }, //any value you want to test
};

(async () => {
  const pingTest = await pingWebhook(webhookData);
  console.log(pingTest);
})();
